cluster-autoscaler/core/scaledown/longestevaluationtracker/longest_node_evaluation_tracker.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package longestevaluationtracker
+
+import (
+	"time"
+
+	"k8s.io/autoscaler/cluster-autoscaler/metrics"
+)
+
+// LongestNodeScaleDownEvalTime is a time tracker for the longest evaluation time of a node during ScaleDown
+type LongestNodeScaleDownEvalTime struct {
+	// lastEvalTime is the time of previous currentlyUnneededNodeNames parsing
+	lastEvalTime time.Time
+	// NodeNamesWithTimeStamps is maps of nodeNames with their time of last successful evaluation
+	NodeNamesWithTimeStamps map[string]time.Time
```

Comment:
Why is this public? This should be an implementation detail. Ideally tests would just evaluate whether return value from `Update` is correct.